lib/arkana/templates/kotlin/arkana.kt.erb

```diff
@@ -0,0 +1,59 @@
+<% require "arkana/helpers/string" %>
+<% require "arkana/helpers/kotlin_template_helper" %>
+<% # TODO: Sort these import statements alphabetically %>
+// DO NOT MODIFY
+// Automatically generated by Arkana (https://github.com/rogerluan/arkana)
+package <%= @kotlin_package_name %>
+
+
+object <%= @namespace %> {
+    private val salt = listOf(<%= @salt.formatted %>)
+
+    internal fun decode(encoded: List<Int>, cipher: List<Int>): String {
+        val decoded = encoded.mapIndexed { index, item ->
+            (item xor cipher[(index % cipher.size)]).toByte()
+        }.toByteArray()
+        return decoded.toString(Charsets.UTF_8)
+    }
+
+    internal fun decodeInt(encoded: List<Int>, cipher: List<Int>): Int {
+        return decode(encoded = encoded, cipher = cipher).toInt()
+    }
+
+    internal fun decodeBoolean(encoded: List<Int>, cipher: List<Int>): Boolean {
+        return decode(encoded = encoded, cipher = cipher).toBoolean()
+    }
+
+    object Global {
+<% @global_secrets.each_with_index do |secret, index| %>
+        val <%= secret.key.camel_case %>: <%= KotlinTemplateHelper.kotlin_type(secret.type) %>
```

Comment:
In Swift I decided to allow users to decide which type of variable declaration they'd like to use (`lazy var` - a lazily instantiated variable, `var` - a variable, `let` - a constant). Do you think this would also be valuable here?

No need to make any changes in this PR regarding this, just wondering if in Android it would make any sense to use other types of variable declarations, for potential future improvements :)

Reply:
I think that might be an overkill here, lazy evaluation adds more overhead in kotlin for trivial evaluations. It may not be appropriate for this use case.

Reply:
We can always allow the user the option to enable lazy properties in the future but as @humblerookie has mentioned, `lazy` properties in Kotlin use [delegates](https://kotlinlang.org/docs/delegated-properties.html), which could potentially add more overhead than repeated calls to the `decode` methods.

Reply:
That's good to know! I trust your experience 💪 thanks for the input! ❤️